src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificatePal.Windows.PrivateKey.cs

```diff
@@ -188,8 +194,51 @@ public ICertificatePal CopyWithPrivateKey(ECDiffieHellman ecdh)
 
         public ICertificatePal CopyWithPrivateKey(MLDsa privateKey)
         {
-            throw new PlatformNotSupportedException(
-                SR.Format(SR.Cryptography_AlgorithmNotSupported, nameof(MLDsa)));
+            if (privateKey is MLDsaCng mldsaCng)
+            {
+                CngKey key = mldsaCng.GetCngKey();
+
+                ICertificatePal? clone = CopyWithPersistedCngKey(key);
+
+                if (clone != null)
+                {
+                    return clone;
+                }
+
+                // TODO why are the other keys (ECDsa, Dsa, etc.) copying via export/import
+                // instead of doing the following? Maybe this fails if key isn't exportable
+                // (the export/import method can handle it by exporting encrypted pkcs#8)?
+                using (MLDsaCng clonedKey = new MLDsaCng(key))
+                {
+                    return CopyWithEphemeralKey(clonedKey.GetCngKey());
```

Comment:
What role does `clonedKey` serve here?  It's going to be functionally the same as `mldsaCng`, and `clonedKey.GetCngKey()` will be the same as `key`.